lib/PeerRPCProxy.js

```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+const Base = require('grenache-nodejs-base')
+const GrWs = require('grenache-nodejs-ws')
+const GrHttp = require('grenache-nodejs-http')
+
+class StatefulPeerRPCProxy extends Base.PeerRPCServer {
+  constructor (link, conf = {}) {
+    super(link, conf)
+    this.proxiedServices = new Map()
+  }
+
+  init () {
+    super.init()
+    switch (this.conf.proxyType.toLowerCase()) {
+      case 'ws':
+        this.server = new GrWs.PeerRPCServer(this.link, this.conf)
+        break
+      case 'http':
+        this.server = new GrHttp.PeerRPCServer(this.link, this.conf)
+        break
+      default:
+        throw new Error('Unsupported proxy type: ' + this.conf.proxyType)
+    }
+    this.server.init()
+  }
+
+  transport (dest, opts = {}) {
+    let t = this.server.transport(dest, opts)
+    t.on('request', (rid, key, payload, handler) => {
+      if (payload) {
```

Comment:
@fyang1024 I usually prefer 
```
if (!payload) return
```
so we can save a bit of indentation

Reply:
good point